From f398c7cbca15cdd264e72e2f024248089ee3574b Mon Sep 17 00:00:00 2001
From: Tatsuya Matsuo <96454784+matsuo-tat@users.noreply.github.com>
Date: Tue, 16 Jan 2024 17:34:35 +0900
Subject: [PATCH 6/7] [xcvr] Limit the VDM items for performance

This change limits the VDM items. This reduces processing time to get the
VDM items and improve performance.
---
 .../sonic_xcvr/codes/public/cmis.py           | 85 ++++++++++---------
 1 file changed, 43 insertions(+), 42 deletions(-)

diff --git a/sonic_platform_base/sonic_xcvr/codes/public/cmis.py b/sonic_platform_base/sonic_xcvr/codes/public/cmis.py
index b757635..c05f59f 100644
--- a/sonic_platform_base/sonic_xcvr/codes/public/cmis.py
+++ b/sonic_platform_base/sonic_xcvr/codes/public/cmis.py
@@ -70,52 +70,53 @@ class CmisCodes(Sff8024):
         12: 'ConfigInProgress',
     }
 
+    # Limit the VDM items to reduce the delay in state retrieval.
     VDM_TYPE = {
         # VDM_ID: [VDM_NAME, DATA_TYPE, SCALE]
-        1: ['Laser Age [%]', 'U16', 1],
-        2: ['TEC Current [%]', 'S16', 100.0/32767],
-        3: ['Laser Frequency Error [MHz]', 'S16', 10],
-        4: ['Laser Temperature [C]', 'S16', 1.0/256],
-        5: ['eSNR Media Input [dB]', 'U16', 1.0/256],
-        6: ['eSNR Host Input [dB]', 'U16', 1.0/256],
-        7: ['PAM4 Level Transition Parameter Media Input [dB]', 'U16', 1.0/256],
-        8: ['PAM4 Level Transition Parameter Host Input [dB]', 'U16', 1.0/256],
-        9: ['Pre-FEC BER Minimum Media Input', 'F16', 1],
-        10: ['Pre-FEC BER Minimum Host Input', 'F16', 1],
-        11: ['Pre-FEC BER Maximum Media Input', 'F16', 1],
-        12: ['Pre-FEC BER Maximum Host Input', 'F16', 1],
+        # 1: ['Laser Age [%]', 'U16', 1],
+        # 2: ['TEC Current [%]', 'S16', 100.0/32767],
+        # 3: ['Laser Frequency Error [MHz]', 'S16', 10],
+        # 4: ['Laser Temperature [C]', 'S16', 1.0/256],
+        # 5: ['eSNR Media Input [dB]', 'U16', 1.0/256],
+        # 6: ['eSNR Host Input [dB]', 'U16', 1.0/256],
+        # 7: ['PAM4 Level Transition Parameter Media Input [dB]', 'U16', 1.0/256],
+        # 8: ['PAM4 Level Transition Parameter Host Input [dB]', 'U16', 1.0/256],
+        # 9: ['Pre-FEC BER Minimum Media Input', 'F16', 1],
+        # 10: ['Pre-FEC BER Minimum Host Input', 'F16', 1],
+        # 11: ['Pre-FEC BER Maximum Media Input', 'F16', 1],
+        # 12: ['Pre-FEC BER Maximum Host Input', 'F16', 1],
         13: ['Pre-FEC BER Average Media Input', 'F16', 1],
-        14: ['Pre-FEC BER Average Host Input', 'F16', 1],
-        15: ['Pre-FEC BER Current Value Media Input', 'F16', 1],
-        16: ['Pre-FEC BER Current Value Host Input', 'F16', 1],
-        17: ['Errored Frames Minimum Media Input', 'F16', 1],
-        18: ['Errored Frames Minimum Host Input', 'F16', 1],
-        19: ['Errored Frames Maximum Media Input', 'F16', 1],
-        20: ['Errored Frames Maximum Host Input', 'F16', 1],
-        21: ['Errored Frames Average Media Input', 'F16', 1],
-        22: ['Errored Frames Average Host Input', 'F16', 1],
-        23: ['Errored Frames Current Value Media Input', 'F16', 1],
-        24: ['Errored Frames Current Value Host Input', 'F16', 1],
-        128: ['Modulator Bias X/I [%]', 'U16', 100.0/65535],
-        129: ['Modulator Bias X/Q [%]', 'U16', 100.0/65535],
-        130: ['Modulator Bias Y/I [%]', 'U16', 100.0/65535],
-        131: ['Modulator Bias Y/Q [%]', 'U16', 100.0/65535],
-        132: ['Modulator Bias X_Phase [%]', 'U16', 100.0/65535],
-        133: ['Modulator Bias Y_Phase [%]', 'U16', 100.0/65535],
-        134: ['CD high granularity, short link [ps/nm]', 'S16', 1],
-        135: ['CD low granularity, long link [ps/nm]', 'S16', 20],
-        136: ['DGD [ps]', 'U16', 0.01],
-        137: ['SOPMD [ps^2]', 'U16', 0.01],
-        138: ['PDL [dB]', 'U16', 0.1],
-        139: ['OSNR [dB]', 'U16', 0.1],
-        140: ['eSNR [dB]', 'U16', 0.1],
-        141: ['CFO [MHz]', 'S16', 1],
-        142: ['EVM_modem [%]', 'U16', 100.0/65535],
+        # 14: ['Pre-FEC BER Average Host Input', 'F16', 1],
+        # 15: ['Pre-FEC BER Current Value Media Input', 'F16', 1],
+        # 16: ['Pre-FEC BER Current Value Host Input', 'F16', 1],
+        # 17: ['Errored Frames Minimum Media Input', 'F16', 1],
+        # 18: ['Errored Frames Minimum Host Input', 'F16', 1],
+        # 19: ['Errored Frames Maximum Media Input', 'F16', 1],
+        # 20: ['Errored Frames Maximum Host Input', 'F16', 1],
+        # 21: ['Errored Frames Average Media Input', 'F16', 1],
+        # 22: ['Errored Frames Average Host Input', 'F16', 1],
+        # 23: ['Errored Frames Current Value Media Input', 'F16', 1],
+        # 24: ['Errored Frames Current Value Host Input', 'F16', 1],
+        # 128: ['Modulator Bias X/I [%]', 'U16', 100.0/65535],
+        # 129: ['Modulator Bias X/Q [%]', 'U16', 100.0/65535],
+        # 130: ['Modulator Bias Y/I [%]', 'U16', 100.0/65535],
+        # 131: ['Modulator Bias Y/Q [%]', 'U16', 100.0/65535],
+        # 132: ['Modulator Bias X_Phase [%]', 'U16', 100.0/65535],
+        # 133: ['Modulator Bias Y_Phase [%]', 'U16', 100.0/65535],
+        # 134: ['CD high granularity, short link [ps/nm]', 'S16', 1],
+        # 135: ['CD low granularity, long link [ps/nm]', 'S16', 20],
+        # 136: ['DGD [ps]', 'U16', 0.01],
+        # 137: ['SOPMD [ps^2]', 'U16', 0.01],
+        # 138: ['PDL [dB]', 'U16', 0.1],
+        # 139: ['OSNR [dB]', 'U16', 0.1],
+        # 140: ['eSNR [dB]', 'U16', 0.1],
+        # 141: ['CFO [MHz]', 'S16', 1],
+        # 142: ['EVM_modem [%]', 'U16', 100.0/65535],
         143: ['Tx Power [dBm]', 'S16', 0.01],
-        144: ['Rx Total Power [dBm]', 'S16', 0.01],
-        145: ['Rx Signal Power [dBm]', 'S16', 0.01],
-        146: ['SOP ROC [krad/s]', 'U16', 1],
-        147: ['MER [dB]', 'U16', 0.1]
+        # 144: ['Rx Total Power [dBm]', 'S16', 0.01],
+        # 145: ['Rx Signal Power [dBm]', 'S16', 0.01],
+        # 146: ['SOP ROC [krad/s]', 'U16', 1],
+        # 147: ['MER [dB]', 'U16', 0.1]
     }
 
     CDB_FAIL_STATUS = {
-- 
2.20.1

